### ISSUE TITLE
Breaking change: Passing .editorconfig to `AdditionalFiles` is no longer supported

### ISSUE THREAD

**Issue by Youssef1313** [dotnet/docs#24894]
## Passing .editorconfig to `AdditionalFiles` is no longer supported

Roslyn analyzers added support for parsing and respecting `.editorconfig` options prior to compiler adding support in VS2019 16.3. This was done via passing `.editorconfig` files as additional files.

### Version introduced

TBD

### Old behavior

Passing `.editorconfig` to `AdditionalFiles` works.

### New behavior

Passing `.editorconfig` to `AdditionalFiles` is no longer supported.

### Reason for change

The compiler added support for parsing `.editorconfig`s in VS2019 16.3. In most cases, you don't need to pass `.editorconfig` to `AdditionalFiles` and it will work. 

### Recommended action

If you're on an older version of Visual Studio, we recommend that you update to a version later than 16.3.

If you're on a version later than 16.3, you shouldn't pass `.editorconfig` to `AdditionalFiles` if the file is in the project's directory or any ancestor directory. Otherwise, you might want to convert it to a `globalconfig`.

### Category

- [ ] ASP.NET Core
- [ ] C#
- [x] Code analysis
- [ ] Core .NET libraries
- [ ] Cryptography
- [ ] Data
- [ ] Debugger
- [ ] Deployment
- [ ] Globalization
- [ ] Interop
- [ ] JIT
- [ ] LINQ
- [ ] Managed Extensibility Framework (MEF)
- [ ] MSBuild
- [ ] Networking
- [ ] Printing
- [ ] SDK
- [ ] Security
- [ ] Serialization
- [ ] Visual Basic
- [ ] Windows Forms
- [ ] Windows Presentation Foundation (WPF)
- [ ] XML, XSLT

### Affected APIs

Not detectable via API analysis

---
#### Issue metadata

* Issue type: breaking-change

**Comment by Youssef1313** [comment]
@mavasani @gewarren

**Comment by Youssef1313** [comment]
Product change PR: https://github.com/dotnet/roslyn-analyzers/pull/5066.

**Comment by gewarren** [comment]
I'm waiting on an SDK version for this one before I can document it.

**Comment by Youssef1313** [comment]
Ping @mavasani for few missing information.

**Comment by Youssef1313** [comment]
@sharwell IIRC I did that per @mavasani's review. I think the fix should be as simple as removing this:

https://github.com/dotnet/roslyn-analyzers/blob/086d1cc888ca0d8c96b94444935f488fcaf3fe98/src/Utilities/Compiler/Options/AnalyzerOptionsExtensions.cs#L590-L593

and update the docs in dotnet/roslyn-analyzers repo as well as dotnet/docs.

**Comment by stuartleal** [comment]
Hi all! 

I'm wondering about this check. I've upgraded from VS 2019 to VS 2022, and a project in my team now has a bunch of AD00001 warnings related to this.

At first, I saw this post: https://github.com/dotnet/roslyn-analyzers/issues/5725 , and thought the issue was having both VS 2019 and VS 2022 on the same machine. I've reinstalled a fresh copy of windows after struggling to get rid of all VS traces, and installed a fresh copy of VS 2022, the latest version. But I still see the AD00001 errors. 

I'm looking for a solution that doesn't involve modifying project code, since there are folks in my team that are still using VS 2019. Before I was using VS 2019 16.17, so I don't know what's happening here.

Can someone point me in the right direction? Thanks.

**Comment by Bosch-Eli-Black** [comment]
@sharwell Would it make sense for any `.editorconfig` files that are detected by the IDE/compiler to be automatically passed to build-time analyzers? As an end user, if the IDE/compiler automatically detects `.editorconfig` files, then I'd prefer not to have to also add them as `AdditionalFiles` just so that they can be processed by analyzers 🙂

**Comment by Youssef1313** [comment]
@Bosch-Eli-Black This is already the case. If `.editorconfig` files are detected then it will be used. That's why this breaking change was introduced.

The only case where `.editorconfig` won't be automatically detected is if it's referenced from a non-ancestor directory. For example if `C:\MyProject\MyProject.csproj` is referencing editorconfig from `C:\Test\.editorconfig`, there is no way we can include it automatically. This is an uncommon case, that's why I was suggesting removing the exception entirely (https://github.com/dotnet/docs/issues/24894#issuecomment-973095929)

**Comment by Bosch-Eli-Black** [comment]
Thanks, @Youssef1313 ! 🙂

My comment was mostly in reply to https://github.com/dotnet/docs/issues/24894#issuecomment-973056421 . Is that comment still correct?

**Comment by Youssef1313** [comment]
@Bosch-Eli-Black Oh sorry I misunderstood your comment.

I think yes, https://github.com/dotnet/docs/issues/24894#issuecomment-973056421 is still relevant and correct (especially for the uncommon scenario I demonstrated earlier).

However, for the very common scenario, I wonder if `AnalyzerConfigOptions` can work and be sufficient. But @sharwell will definitely know better :)

**Comment by Bosch-Eli-Black** [comment]
@sharwell or @Youssef1313 , Can this please be reopened? It seems there's still work to do here 🙂

**Comment by Bosch-Eli-Black** [comment]
(Or should I open a new issue somewhere to suggest that this warning be removed, re the comments above: https://github.com/dotnet/docs/issues/24894#issuecomment-973056421 and https://github.com/dotnet/docs/issues/24894#issuecomment-973095929 ?)

**Comment by Bosch-Eli-Black** [comment]
@Youssef1313 Oh, cool! Thanks 🙂
